src/wallet/rpcwallet.cpp

```diff
@@ -3284,6 +3292,29 @@ Value z_sendmany(const Array& params, bool fHelp)
         }
     }
 
+    // Check the number of zaddr outputs does not exceed the limit.
+    if (zaddrRecipients.size() > Z_SENDMANY_MAX_ZADDR_OUTPUTS)  {
```

Comment:
What if a change address takes this over the limit? Are there other ways the underlying API might add data to the transaction that wouldn't be accounted for in these changes?